### ISSUE TITLE
Crash when user dictionary contains Date

### ISSUE THREAD

**Issue by JanGorman** [amplitude/experiment-ios-client#42]
The experiment client crashes encoding JSON (on this line https://github.com/amplitude/experiment-ios-client/blob/main/Sources/Experiment/ExperimentClient.swift#L482) if the user dictionary contains a date:

```
*** Terminating app due to uncaught exception 'NSInvalidArgumentException', reason: 'Invalid type in JSON write (__NSTaggedDate)'
*** First throw call stack:
(0x194afa69c 0x18cd8fc80 0x1939f2c70 0x1939f25c8 0x18e1251b0 0x18dea0ba0 0x1939f20c4 0x1939f25c8 0x18e1251b0 0x18dea0ba0 0x1939f20c4 0x193a26074 0x193a25f1c 0x10162c928 0x10162a1d4 0x101625e88 0x1015f8258 0x10cb9cf50 0x10cb9eb34 0x10cba698c 0x10cba7728 0x10cbb45f8 0x10cbb3c2c 0x1fea33964 0x1fea33a04)
libc++abi: terminating due to uncaught exception of type NSException
```

**Comment by tyiuhc** [comment]
@JanGorman, thanks for raising this issue. The date property should be passed as a string to avoid this issue.

**Comment by JanGorman** [comment]
That's what we ended up doing of course but it comes rather surprising and it's never cool if a third party crashes your code.

**Comment by daltonclaybrook** [comment]
My team is experiencing this issue as well. Has a fix for this been prioritized within Amplitude @tyiuhc?

**Comment by tyiuhc** [comment]
@JanGorman @daltonclaybrook, thank you for your patience on this. A fix for this is currently being worked on.

**Comment by tyiuhc** [comment]
@JanGorman @daltonclaybrook, this issue has been addressed in v1.13.5, please let us know if it is not behaving as expected, thanks!
